### DISCUSSION TITLE
Reset Error on Field onFocus

### DISCUSSION THREAD

**Discussion by defaultgnome** [TanStack/form#530] [Q&A]
I want to reset the error on the focused field.
is there a built-in way to do that with tanstack/form, or i need to manage this state myself for each field?

**Comment by crutchcorn** [comment]
Just curious; what's the benefit in doing so? I'm not sure I understand what the usecase would be

**Comment by n3ps** [comment]
A possible use case:
1. `onBlur` validation - triggers an error
2. `onFocus` user should not keep seeing the error - give them an opportunity to change the value first

**Reply by tomatau** [reply]
Similar use case here:
1. `onBlurAsync` validates for uniqueness on the server
2. `onChange` should be able to clear the blurred error state as the user types
3. `onBlurAsync` triggered again with the new value

Pushing all the validation into `onChangeAsync` with `onChangeAsyncDebounceMs` isn't always ideal here if the check is expensive
